exist-core/src/main/java/org/exist/xquery/functions/fn/FunXmlToJson.java

```diff
@@ -0,0 +1,246 @@
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ *  $Id$
+ */
+package org.exist.xquery.functions.fn;
+
+import com.fasterxml.jackson.core.*;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.exist.dom.QName;
+import org.exist.xquery.*;
+import org.exist.xquery.functions.map.MapType;
+import org.exist.xquery.value.*;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamReader;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.io.Writer;
+import java.util.Stack;
+
+/**
+ * @author <a href="mailto:from-github-existdb@agh2342.de">Adrian Hamm</a>
+ */
+public class FunXmlToJson extends BasicFunction {
+
+    protected static final Logger logger = LogManager.getLogger(FunXmlToJson.class);
+
+    public final static FunctionSignature[] signature = {
+            new FunctionSignature(
+                    new QName("xml-to-json", Function.BUILTIN_FUNCTION_NS),
+                    "Converts an XML tree (in w3c 'XML Representation of JSON' format) into a string conforming to the JSON grammar. Basic string (un)escaping.",
+                    new SequenceType[]{
+                            new FunctionParameterSequenceType("node", Type.NODE,
+                                    Cardinality.ZERO_OR_ONE, "The input node")
+                    },
+                    new FunctionReturnSequenceType(Type.STRING, Cardinality.ZERO_OR_ONE,
+                            "The JSON representation of the input node")
+            ),
+            new FunctionSignature(
+                    new QName("xml-to-json", Function.BUILTIN_FUNCTION_NS),
+                    "Converts an XML tree (in w3c 'XML Representation of JSON' format) into a string conforming to the JSON grammar. Basic string (un)escaping. Options are ignored.",
+                    new SequenceType[]{
+                            new FunctionParameterSequenceType("node", Type.NODE,
+                                    Cardinality.ZERO_OR_ONE, "The input node"),
+                            new FunctionParameterSequenceType("options", Type.MAP,
+                                    Cardinality.ONE, "The options map")
+                    },
+                    new FunctionReturnSequenceType(Type.STRING, Cardinality.ZERO_OR_ONE,
+                            "The JSON representation of the input node")
+            )
+    };
+
+    public FunXmlToJson(XQueryContext context, FunctionSignature signature) { super(context, signature); }
+
+    public Sequence eval(Sequence[] args, Sequence contextSequence) throws XPathException {
+        Sequence result;
+        final Sequence seq = (getArgumentCount() > 0) ? args[0] : Sequence.EMPTY_SEQUENCE;
+        //TODO: implement handling of options
+        final MapType options = (getArgumentCount() > 1) ? (MapType) args[1].itemAt(0) : new MapType(context);
+
+        if (seq.isEmpty())
+            {result = Sequence.EMPTY_SEQUENCE;}
```

Comment:
Can you fix the formatting to be consistent please? I usually just use the IntelliJ format code option.